### DISCUSSION TITLE
Can't get Storybook to work with CSS Modules using SCSS

### DISCUSSION THREAD

**Discussion by xander-design** [storybookjs/storybook#26103] [Help]
Hi, 

I'm having problems getting Storybook to work with my react.js project when I'm using CSS Modules which contain SCSS. Below is my webpack.copy.js file which works fine when I run "npm start".

```js
const paths = require("../../paths");
const HtmlWebpackPlugin = require("html-webpack-plugin");
const MiniCssExtractPlugin = require("mini-css-extract-plugin");
const { presets, plugins } = require("./webpack.config.babel");

module.exports = {
  mode: "development",
  entry: paths.fromRoot("src/index.js"),
  output: {
    path: paths.fromRoot("build"),
    filename: "[name].[hash].js",
  },
  resolve: {
    extensions: [".js", ".jsx"],
  },
  module: {
    rules: [
      {
        test: /\.(js|jsx)$/,
        exclude: /node_modules/,
        use: {
          loader: "babel-loader",
          options: {
            babelrc: false,
            presets: presets,
            plugins: plugins,
          },
        },
      },
      {
        test: /\.html$/,
        use: [{ loader: "html-loader" }],
      },
      {
        test: /\.svg$/,
        use: ["@svgr/webpack"],
      },
      {
        test: /\.(s?css)$/,
        use: [
          MiniCssExtractPlugin.loader,
          {
            loader: "css-loader",
            options: {
              importLoaders: 1,
              modules: {
                localIdentName: "[name]__[local]___[hash:base64:5]",
              },
            },
          },
          "sass-loader",
        ],
      },
    ],
  },
  plugins: [
    new MiniCssExtractPlugin({
      filename: "[name].[hash].css",
    }),
    new HtmlWebpackPlugin({
      template: paths.fromRoot("src/index.html"),
      filename: "./index.html",
      inject: "body",
    }),
  ],
  devServer: {
    contentBase: paths.fromRoot("build"),
    hot: true,
    open: true,
  },
};
```

and below is a copy of my current Storybook main.js file:
```js
const path = require("path");

module.exports = {
  addons: [
    "storybook-readme/register",
    "@storybook/addon-a11y/register",
    "@storybook/addon-actions/register",
    "@storybook/addon-backgrounds/register",
    "@storybook/addon-docs/register",
    "@storybook/addon-docs/preset",
    "@storybook/addon-events/register",
    "@storybook/addon-jest/register",
    "@storybook/addon-knobs/register",
    "@storybook/addon-links/register",
    "@storybook/addon-viewport/register",
  ],

  stories: ["../src/**/*.stories.[tj]s"],

  webpackFinal: async (config, { configType }) => {
    // get index of css rule
    const ruleCssIndex = config.module.rules.findIndex(
      (rule) => rule.test.toString() === "/\\.css$/"
    );

    // map over the 'use' array of the css rule and set the 'module' option to true
    config.module.rules[ruleCssIndex].use.map((item) => {
      if (item.loader && item.loader.includes("/css-loader/")) {
        item.options.modules = {
          mode: "local",
          localIdentName:
            configType === "PRODUCTION"
              ? "[local]__[hash:base64:5]"
              : "[name]__[local]__[hash:base64:5]",
        };
      }

      return item;
    });

    config.module.rules.push({
      test: /\.scss$/,
      use: ["style-loader", "css-loader", "sass-loader"],
      include: path.resolve(__dirname, "../"),
    });

    return config;
  },
};
```
When I run storybook
```js
npm run storybook
```
The Storybook browser window opens but the component has no css styles applied. I can only assume that the css file is not being created from the scss because the Storybook main.js config is wrong.

Can anyone help please...

**Comment by stale** [comment]
Hi everyone! Seems like there hasn't been much going on in this issue lately. If there are still questions, comments, or bugs, please feel free to continue the discussion. Unfortunately, we don't have time to get to every issue. We are always open to contributions so please send us a pull request if you would like to help. Inactive issues will be closed after 30 days. Thanks!

**Comment by CraftingGamerTom** [comment]
I had the same issue. For me, it required a few changes. I hope this is relevant to you :)

## Steps to fix

1) 
I didnt have sass installed 🤦 
So I ran `npm install sass --save`

2) 
The code used to push scss rules seemed to be causing issues. Definetly wasnt helping. Sorry I don't have more information for 'why' but **REMOVE** the code below from your `main.js`
```
// REMOVE THIS

config.module.rules.push({
  test: /\.scss$/,
  use: ["style-loader", "css-loader", "sass-loader"],
  include: path.resolve(__dirname, "../"),
});

// REMOVE THIS
```

3) 
Use a storybook addon for scss 
Install the addon by running
`npm install @storybook/preset-scss --save`

Then add the following addon to your list of used addons
`'@storybook/preset-scss',`


## Summary
OPs `main.js` should be updated to look like this
```
const path = require("path");

module.exports = {
  addons: [
    "storybook-readme/register",
    "@storybook/addon-a11y/register",
    "@storybook/addon-actions/register",
    "@storybook/addon-backgrounds/register",
    "@storybook/addon-docs/register",
    "@storybook/addon-docs/preset",
    "@storybook/addon-events/register",
    "@storybook/addon-jest/register",
    "@storybook/addon-knobs/register",
    "@storybook/addon-links/register",
    "@storybook/addon-viewport/register",
    "@storybook/preset-scss",
  ],

  stories: ["../src/**/*.stories.[tj]s"],

  webpackFinal: async (config, { configType }) => {
    // get index of css rule
    const ruleCssIndex = config.module.rules.findIndex(
      (rule) => rule.test.toString() === "/\\.css$/"
    );

    // map over the 'use' array of the css rule and set the 'module' option to true
    config.module.rules[ruleCssIndex].use.map((item) => {
      if (item.loader && item.loader.includes("/css-loader/")) {
        item.options.modules = {
          mode: "local",
          localIdentName:
            configType === "PRODUCTION"
              ? "[local]__[hash:base64:5]"
              : "[name]__[local]__[hash:base64:5]",
        };
      }

      return item;
    });

    return config;
  },
};
```


## References
https://storybook.js.org/docs/react/api/presets

https://github.com/storybookjs/presets/tree/master/packages/preset-scss

**Comment by CraftingGamerTom** [comment]
@iamxanderg I hope my above comment helps. Sorry for the double comment, I should have tagged you in my first response.

**Comment by stale** [comment]
Hi everyone! Seems like there hasn't been much going on in this issue lately. If there are still questions, comments, or bugs, please feel free to continue the discussion. Unfortunately, we don't have time to get to every issue. We are always open to contributions so please send us a pull request if you would like to help. Inactive issues will be closed after 30 days. Thanks!

**Comment by qkreltms** [comment]
Still same in `"@storybook/react": "6.1.18 or 6.1.17"` and `"@storybook/preset-scss": 1.0.3`
I installed storybook using `npx sb init` and also tried @CraftingGamerTom's way, but doesn't works.
Implemented example https://github.com/qkreltms/lerna-test1/tree/others

**Comment by RodrigoTomeES** [comment]
> 
> 
> Still same in `"@storybook/react": "6.1.18 or 6.1.17"` and `"@storybook/preset-scss": 1.0.3`
> I installed storybook using `npx sb init` and also tried @CraftingGamerTom's way, but doesn't works.
> Implemented example https://github.com/qkreltms/lerna-test1/tree/others

Hi, I had the same problem but I was able to fix it by changing the sass-loader version from 11 to 10. I have used the @storybook/preset-scss.

https://stackoverflow.com/questions/66082397/typeerror-this-getoptions-is-not-a-function

**Comment by gwithian** [comment]
Hi, I managed to get SCSS modules working by adding the following config to the `webpackFinal` in `.storybook/main.js`.
This solution uses no add-ons/presets, just needed `sass` and `sass-loader@10`(see above - v11 didn't work) modules installed.
My use case is fairly simple (using `Component.module.scss` type files with my components) but might help others.

```JavaScript
webpackFinal: async (config) => {
    // add SCSS support for CSS Modules
    config.module.rules.push({
      test: /\.scss$/,
      use: ['style-loader', 'css-loader?modules&importLoaders', 'sass-loader'],
      include: path.resolve(__dirname, '../'),
    });

    return config;
}
```

In my react app code I am using styles object like so:

```jsx
import React from 'react';
import styles from './Component.module.scss';

function Component() {
  return (
    <div className={styles.wrapper}>
     foobar
    </div>
  );
}

export default Component;
```

**Comment by rpedroni** [comment]
@gwithian's solution is the only that worked for me. Special attention to use `sass-loader@10` 👍

**Comment by baileydoestech** [comment]
Above also worked for us (sass-loader @10), I think the issue is that sass-loader >11 minimum webpack version is now 5

https://github.com/webpack-contrib/sass-loader/blob/v11.1.1/CHANGELOG.md#1100-2021-02-05

**Comment by czabaj** [comment]
The `@storybook/preset-scss` [allows override the `rule` webpack config](https://github.com/storybookjs/presets/blob/master/packages/preset-scss/index.js), so this is my solution

```JavaScript
// .storybook/main.js

const config = {
     addons: [
		...
		// nomodule
		{
			name: `@storybook/preset-scss`,
			options: {
				rule: {
					test: /(?<!\.module).s[ca]ss$/,
				}
			},
		},
		// module
		{
			name: `@storybook/preset-scss`,
			options: {
				rule: {
					test: /\.module\.s[ca]ss$/,
				},
				cssLoaderOptions: {
					modules: {
						localIdentName: '[name]__[local]--[hash:base64:5]',
					},
				}
			},
		},
	],
}
```

**Comment by ericrinaldo** [comment]
This worked for me:
- Updated packages
- No addon or preset
- Mix between modules and non-module scss files

dependency versions:
```
"@storybook/react": "^6.4.0-rc.3",
"css-loader": "6.5.1",
"sass": "^1.43.4",
"sass-loader": "^12.3.0",
"style-loader": "2.0.0",
"webpack": "5.31.2",
```


```
// .storybook/main.js

const path = require('path');

module.exports = {
  ...
  webpackFinal: async (config) => {
    config.module.rules.push(
      {
        test: /\.s(a|c)ss$/,
        include: path.resolve(__dirname, '../'),
        use: [
          'style-loader',
          {
            loader: 'css-loader',
            options: {
              modules: {
                auto: true,
                localIdentName: '[name]__[local]--[hash:base64:5]',
              },
            },
          },
          'sass-loader'
        ],
      },
    );
    return config;
  },
  ...
};

```


```
// Button.jsx

import React from 'react';
import styles from './Button.module.scss';

export const Button = () => (
  <button className={styles.button}}>
    button
  </button>
);

```


```
// Button.module.scss

.button {
  background: green;
}
```


```
<!-- storybook's iframe -->

<button class="Button-module__button--b_aG7">
  button
</button>
```

**Comment by lzacharkow** [comment]
@ericrinaldo You saved my day!!! Yours is the only solution that worked for me. Special thanks for listing your dependency versions, they're the same as mine.

**Comment by AtAFork** [comment]
@ericrinaldo thank you! After a long struggle, a combination of your response and these Stack Overflow questions helped:

https://stackoverflow.com/questions/68580273/type-error-this-getoptions-is-not-a-function-for-style-loader/68580324#68580324

https://stackoverflow.com/questions/66082397/typeerror-this-getoptions-is-not-a-function/66087132#66087132

Other than some slight tweaks to dependencies, I got it working with @ericrinaldo's code. My dependencies:

```
    "webpack": "^5.61.0"
    "@react-theming/storybook-addon": "^1.1.3",
    "@storybook/addon-essentials": "6.4.0",
    "@storybook/addon-links": "6.4.0",
    "@storybook/react": "^6.4.0-rc.3",
    "css-loader": "5.2.7",
    "style-loader": "2.0.0",
    "sass": "^1.43.4",
    "sass-loader": "10.2.0",
```

**Comment by doug-sheridan** [comment]
> @ericrinaldo thank you! After a long struggle, a combination of your response and these Stack Overflow questions helped:
> 
> https://stackoverflow.com/questions/68580273/type-error-this-getoptions-is-not-a-function-for-style-loader/68580324#68580324
> 
> https://stackoverflow.com/questions/66082397/typeerror-this-getoptions-is-not-a-function/66087132#66087132
> 
> Other than some slight tweaks to dependencies, I got it working with @ericrinaldo's code. My dependencies:
> 
> ```
>     "webpack": "^5.61.0"
>     "@react-theming/storybook-addon": "^1.1.3",
>     "@storybook/addon-essentials": "6.4.0",
>     "@storybook/addon-links": "6.4.0",
>     "@storybook/react": "^6.4.0-rc.3",
>     "css-loader": "5.2.7",
>     "style-loader": "2.0.0",
>     "sass": "^1.43.4",
>     "sass-loader": "10.2.0",
> ```

@AtAFork and @ericrinaldo's solution together worked for me, running SB v6.4.9 with Typescript :) Thank you gents!

**Comment by 7iomka** [comment]
> ericrinaldo

Hi.
Thanks, your solution works fine, but I have error when i add postcss-loader
```js

config.module.rules.push({
  test: /\.s(a|c)ss$/,
  use: [
    'style-loader',
    {
      loader: 'css-loader',
      options: {
        importLoaders: 1, // We always need to apply postcss-loader before css-loader
        modules: {
          auto: /\.module\.scss$/, // true
          localIdentName: '[name]__[local]--[hash:base64:5]',
        },
      },
    },
    {
      loader: 'postcss-loader', // required for tailwind
      options: {
        implementation: require('postcss'), // postcss 8
        postcssOptions: {
          config: path.resolve(__dirname, '../postcss.config.js'),
        },
      },
    },
    {
      loader: 'sass-loader',
      options: {
        // sourceMap: true,
        ...sassLoaderOptions,
        implementation: require('sass'), // dart-sass
      },
    },
  ],
});

    
```

Error occured
```
ModuleBuildError: Module build failed (from ./node_modules/postcss-loader/dist/cjs.js):
Error: PostCSS plugin postcss-nested requires PostCSS 8.
Migration guide for end-users:
https://github.com/postcss/postcss/wiki/PostCSS-8-for-end-users
    at Processor.normalize (XXX/node_modules/postcss/lib/processor.js:167:15)
    at new Processor (XXX/node_modules/postcss/lib/processor.js:56:25)
    at Object.postcss [as default] (XXX/node_modules/postcss/lib/postcss.js:55:10)
    at XXX/node_modules/tailwindcss/lib/util/parseObjectStyles.js:21:37
    at Array.flatMap (<anonymous>)
    at parseObjectStyles (XXX/node_modules/tailwindcss/lib/util/parseObjectStyles.js:20:19)
    at Object.parseObjectStyles [as default] (XXX/node_modules/tailwindcss/lib/util/parseObjectStyles.js:16:16)
    at XXX/node_modules/tailwindcss/lib/lib/setupContextUtils.js:102:64
    at Array.flatMap (<anonymous>)
    at parseStyles (XXX/node_modules/tailwindcss/lib/lib/setupContextUtils.js:100:19)
    at withIdentifiers (XXX/node_modules/tailwindcss/lib/lib/setupContextUtils.js:141:12)
    at addComponents (XXX/node_modules/tailwindcss/lib/lib/setupContextUtils.js:305:44)
    at corePlugins.container (XXX/node_modules/tailwindcss/lib/corePlugins.js:291:13)
    at registerPlugins (XXX/node_modules/tailwindcss/lib/lib/setupContextUtils.js:607:61)
    at createContext (XXX/node_modules/tailwindcss/lib/lib/setupContextUtils.js:766:5)
    at Object.getContext (XXX/node_modules/tailwindcss/lib/lib/setupContextUtils.js:821:19)
    at processResult (XXX/node_modules/webpack/lib/NormalModule.js:751:19)
    at XXX/node_modules/webpack/lib/NormalModule.js:853:5
    at XXX/node_modules/loader-runner/lib/LoaderRunner.js:399:11
    at XXX/node_modules/loader-runner/lib/LoaderRunner.js:251:18
    at context.callback (XXX/node_modules/loader-runner/lib/LoaderRunner.js:124:13)
    at Object.loader (XXX/packages/ui/node_modules/postcss-loader/dist/index.js:142:7)
```

I'm tried to
- Add postcss (v8) to peerDependencies - nothing changes
- use addons instead - all have similar or postcss issues

This issue is popular (because tailwind.css is popular too), but any solutions works with css, but not working with sass.
Any help from community, please!

**Comment by ericrinaldo** [comment]
@7iomka, it looks like 'PostCSS plugin postcss-nested' requires PostCSS 8 but Storybook builds with PostCSS 7. (see `npm ls postcss`)

You can force these package dependencies to use PostCSS 8. (Though it might break something else; there might be a reason why the Storybook dependencies use PostCSS 7).

```
// package.json
...
  {
    "resolutions": {
      "**/postcss": "^8.4.4",
  }
...
```

then `yarn`.

**Comment by PaulHaze** [comment]
Hi I have tried the various solutions above and but am still having no luck I get THIS error with Storybook/Tailwind/Nextjs:

```
ModuleBuildError: Module build failed (from ./node_modules/storybook-addon-sass-postcss/node_modules/sass-loader/dist/cjs.js):
SassError: expected "{".
  ╷
2 │       import API from "!../../../node_modules/style-loader/dist/runtime/injectStylesIntoStyleTag.js";
```

Tailwind and my main scss files are appearing no problem, and css modules appear no problem. My current main.js file looks like this (the commented out sections are what seems to cause the problem. It loads fine with them commented out and returns the error above when they are back in)
```
const path = require('path');

module.exports = {
  stories: ['../src/**/*.stories.mdx', '../src/**/*.stories.@(js|jsx|ts|tsx)'],
  staticDirs: ['../public'],
  addons: [
    '@storybook/addon-links',
    '@storybook/addon-essentials',
    '@storybook/addon-interactions',
    // '@storybook/preset-scss',
    {
      name: 'storybook-addon-sass-postcss',
      options: {
        loadSassAfterPostCSS: true,
        postcssLoaderOptions: {
          implementation: require('postcss'),
        },
        rule: {
          test: /\.(scss|sass)$/i,
        },
      },
    },
  ],
  framework: '@storybook/react',
  core: {
    builder: '@storybook/builder-webpack5',
  },
  webpackFinal: (config) => {
    // config.module.rules.push({
    //   test: /\.scss$/,
    //   include: path.resolve(__dirname, '../'),
    //   use: [
    //     'style-loader',
    //     {
    //       loader: 'css-loader',
    //       options: {
    //         modules: {
    //           auto: true,
    //           localIdentName: '[name]__[local]--[hash:base64:5]',
    //         },
    //       },
    //     },
    //     'sass-loader',
    //   ],
    // });
    config.resolve.alias = {
      ...config.resolve?.alias,
      '@': [path.resolve(__dirname, '../src/'), path.resolve(__dirname, '../')],
    };
    config.resolve.roots = [
      path.resolve(__dirname, '../public'),
      'node_modules',
    ];
    return config;
  },
};

```

Does anyone have a solution for this? I feel like I have been spending the last 2 days wrestling with Storybook to get scss modules working...

**Comment by gdiasbruno** [comment]
> This worked for me:
> 
> * Updated packages
> * No addon or preset
> * Mix between modules and non-module scss files
> 
> dependency versions:
> 
> ```
> "@storybook/react": "^6.4.0-rc.3",
> "css-loader": "6.5.1",
> "sass": "^1.43.4",
> "sass-loader": "^12.3.0",
> "style-loader": "2.0.0",
> "webpack": "5.31.2",
> ```
> 
> ```
> // .storybook/main.js
> 
> const path = require('path');
> 
> module.exports = {
>   ...
>   webpackFinal: async (config) => {
>     config.module.rules.push(
>       {
>         test: /\.s(a|c)ss$/,
>         include: path.resolve(__dirname, '../'),
>         use: [
>           'style-loader',
>           {
>             loader: 'css-loader',
>             options: {
>               modules: {
>                 auto: true,
>                 localIdentName: '[name]__[local]--[hash:base64:5]',
>               },
>             },
>           },
>           'sass-loader'
>         ],
>       },
>     );
>     return config;
>   },
>   ...
> };
> ```
> 
> ```
> // Button.jsx
> 
> import React from 'react';
> import styles from './Button.module.scss';
> 
> export const Button = () => (
>   <button className={styles.button}}>
>     button
>   </button>
> );
> ```
> 
> ```
> // Button.module.scss
> 
> .button {
>   background: green;
> }
> ```
> 
> ```
> 
> 
> <button class="Button-module__button--b_aG7">
>   button
> </button>
> ```

This worked for me!

**Comment by azabraao** [comment]
Worked for me to copy'n paste my webpack configurations for sass modules into storybook configs:

```
  webpackFinal: async (config) => {
    // add SCSS support for CSS Modules
    config.module.rules.push({
      test: /\.module\.s(a|c)ss$/,
      use: [
        "style-loader",
        {
          loader: "css-loader",
          options: {
            modules: {
              localIdentName: "[local]__[hash:base64:5]",
              exportLocalsConvention: "camelCase",
            },
            sourceMap: true,
          },
        },
        {
          loader: "sass-loader",
          options: {
            sourceMap: true,
          },
        },
      ],
      include: path.resolve(__dirname, "../"),
    });

    return config;
  },
```

**Comment by jasonrundell** [comment]
My `npm run storybook` works fine with SCSS modules but when I create my library for npm, the scss files are never bundled in the dist folder. Wondering if anyone else came across this issue and found a fix?
My build command is `cross-env BABEL_ENV=production babel src -d dist`
No errors when I build. But when I use my library in another project using npm, there are errors because the .scss files are not inside the component folders or anywhere at all.

**Comment by bpGusar** [comment]
> My `npm run storybook` works fine with SCSS modules but when I create my library for npm, the scss files are never bundled in the dist folder. Wondering if anyone else came across this issue and found a fix? My build command is `cross-env BABEL_ENV=production babel src -d dist` No errors when I build. But when I use my library in another project using npm, there are errors because the .scss files are not inside the component folders or anywhere at all.

have the same issue, have no idea how to fix it (

**Comment by bpGusar** [comment]
> My `npm run storybook` works fine with SCSS modules but when I create my library for npm, the scss files are never bundled in the dist folder. Wondering if anyone else came across this issue and found a fix? My build command is `cross-env BABEL_ENV=production babel src -d dist` No errors when I build. But when I use my library in another project using npm, there are errors because the .scss files are not inside the component folders or anywhere at all.

find solution, works for me
https://www.npmjs.com/package/rollup-plugin-styles#css-modules

**Comment by jasonrundell** [comment]
@bpGusar that looks promising. But my attempt is failing: https://github.com/jasonrundell/dropship/blob/issue/35-build-missing-css-files/rollup.config.js

What's in your .storybook/main.js? Are you using any additional webpack configuration like the solutions posted above?

**Comment by bpGusar** [comment]
@jasonrundell yes, I'm using webpack for dev build and rollup for prod build

in .storybook/main.js I have this webpackFinal config for css modules for dev env

```
...
webpackFinal: async (config) => {
    config.module.rules.push({
      test: /\.s(a|c)ss$/,
      include: path.resolve(__dirname, '../'),
      use: [
        'style-loader',
        {
          loader: 'css-loader',
          options: {
            modules: {
              auto: true,
              localIdentName: [local]__[hash:base64:15]`,
            },
          },
        },
        'sass-loader',
      ],
    });
    return config;
  }
...
```

rollup config rollup.config.js for prod build (dist folder)

```import typescript from 'rollup-plugin-typescript2';
import styles from 'rollup-plugin-styles';
import peerDepsExternal from 'rollup-plugin-peer-deps-external';
import image from '@rollup/plugin-image';
import json from '@rollup/plugin-json';
import stringHash from 'string-hash';
import camelCase from 'lodash/camelCase';
import del from 'rollup-plugin-delete';

export default [
  {
    input: 'src/index.ts',
    output: {
      file: 'dist/index.js',
      format: 'cjs',
      assetFileNames: '[name][extname]',
    },
    plugins: [
      del({ targets: 'dist/*' }),
      peerDepsExternal({
        includeDependencies: true,
      }),
      styles({
        mode: ['extract', './index.css'],
        modules: {
          generateScopedName: (name, filename, css) => {
            if (filename.includes('module')) {
              const hash = stringHash(css).toString(15);
              return `${camelCase(name)}__${hash}`;
            }

            return name;
          },
        },
        url: {
          publicPath: './assets/',
        },
      }),
      typescript({
        tsconfigOverride: {
          exclude: ['**/*.test.tsx', '**/*.test.ts'],
        },
      }),
      image(),
      json(),
    ],
  },
];
```

**Comment by jasonrundell** [comment]
@bpGusar I had no luck with roll-up. I went with Parcel and just followed the basic [library build instructions](https://parceljs.org/getting-started/library/). It worked right away for me. I also had this in my `.storybook/main.js`:

```javascript
  ...
  webpackFinal: async (config) => {
    // Make whatever fine-grained changes you need
    config.module.rules.push({
      test: /\.scss$/,
      use: ['style-loader', 'css-loader', 'sass-loader'],
      include: path.resolve(__dirname, '../src')
    })
    
    // Return the altered config
    return config
  }
```

For my component library, it turns out that having a CSS dependency is not going to be a favourable practice for most modern JS projects (e.g. [NextJS will not build a library with a CSS dependency](https://nextjs.org/docs/messages/css-npm) without some some [module transpilation](https://nextjs.org/docs/advanced-features/compiler#module-transpilation) that I don't want a user to have to go through. So I'm going to ship without any CSS dependencies, as I also don't want a CSS-in-JS approach, and I actually changed my mind around having any opinionated, modular, CSS for my project.

**Comment by Rajaotix** [comment]
This worked on my side
- execute `npx storybook@latest upgrade`
- remove addon `@storybook/preset-scss` if you have it on your addons.
- change the storybook framework to "@storybook/nextjs" (you need to install it not done).

**Comment by valentinpalkovic** [comment]
CSS/SCSS should be configured appropriately for Next.js when the `@storybook/nextjs` framework is used. For the other frameworks which uses Webpack5, I suggest giving this addon a try: https://github.com/storybookjs/addon-styling-webpack. I hope this helps. Otherwise, there are plenty of solutions and workarounds mentioned above. Storybook will not add a kind of default SCSS Module solution because it is impossible to find the right default since user projects are so different. I will transform this issue into a discussion so that people can still follow it. But I don't think it's worth keeping it as an issue.
